### PR TITLE
Implement CICD, Update Testing and Project Structure.

### PR DESCRIPTION
**Pull request by mohamdlog**
**Description:**

This pull request introduces the CI/CD pipeline, improve test coverage, and maintain code consistency across the project. Below is a summary of the changes made:

---

### **CI/CD Pipeline Updates**
1. **Backend Testing Workflow:**
   - Configured Node.js `22.x` environment.
   - Improved dependency management by explicitly caching backend `package-lock.json`.
   - Added environmental variable handling for testing:
     - `MONGODB_URL`, `JWT_SECRET`, `CLIENT_URL`, `SLU_REUSE_SUPPORT_EMAIL`, `SLU_REUSE_SUPPORT_EMAIL_PASSWORD`.
   - Suppressed deprecation warnings using `NODE_OPTIONS`.

2. **Frontend Testing Workflow:**
   - Similar Node.js environment setup for consistency.
   - Cache optimization for frontend tests using `package-lock.json`.
   - Suppressed global deprecation warnings.

---

### **Code Improvements**
1. **Controller Updates:**
   - Resolved capitalization inconsistencies in model imports:
     - `ItemModel.js` → `itemModel.js`.
   - Improved error handling in tests for controllers like `itemController`.

2. **Deleted Unnecessary Files:**
   - Removed `.vscode/settings.json` and redundant test files to reduce clutter.

3. **New Test Coverage:**
   - Added or updated unit tests for critical functionalities:
     - `itemController.test.js`
     - `updateProfile.test.js`
     - `profileRoutes.test.js`

4. **Backend Enhancements:**
   - Refactored `server.js` for improved testability:
     - Conditional database connection and server start based on `NODE_ENV`.

5. **Frontend Adjustments:**
   - Standardized imports for reusable components:
     - Replaced `.mjs` with `.js` extensions in imports.
   - Updated UI text and test assertions for clarity:
     - E.g., `Listings` → `My Listings`.

---

### **Dependency Management**
- Updated `package.json` dependencies and fixed lock file discrepancies.
- Backend:
  - Added `mockingoose` for mock testing Mongoose models.
- Frontend:
  - Updated Jest configurations to transform modules properly.

---

### **Testing Enhancements**
1. **Backend Test Improvements:**
   - Added mock-based tests with `mockingoose` for efficient testing of database interactions.
   - Refactored test cases to cover edge cases like missing tokens, invalid inputs, and database errors.

2. **Frontend Test Improvements:**
   - Updated `App.test.js` to reflect branding changes.
   - Adjusted `ProfileSidebar.test.js` assertions to match updated component labels and routes.

### CODE REVIEW

File: .github/workflows/ci-cd.yml
Comment (shagunsharmaslu) [inline .github/workflows/ci-cd.yml:1]: Your CI/CD pipeline is well-organized, with separate steps for testing the backend and frontend. It securely manages sensitive information using GitHub Secrets and speeds up builds with caching. The pipeline is clear and easy to understand, thanks to labeled steps and specific settings for each part. Suppressing warnings also keeps logs focused on important issues. Overall, it’s a solid setup, ready for additions like code checks or deployment.

File: backend/tests/profile/updateProfile.test.js
Comment (shagunsharmaslu) [inline backend/tests/profile/updateProfile.test.js:1]: Thank you for fixing all the conflicting test classes. I can see you put in some effort here!

**Review by shagunsharmaslu — APPROVED** [review]
Good job. Approving!

**Review by git-voo — APPROVED** [review]
Great! All secrets are now captured by the workflow. Both frontend and backend matrices run successfully

**Pull request merged**